plugins/postcss-cascade-layers/src/desugar-nested-layers.ts

```diff
@@ -0,0 +1,61 @@
+import type { Container, AtRule, ChildNode } from 'postcss';
+import type { Model } from './model';
+import { someAtRuleInTree } from './some-in-tree';
+
+export function desugarNestedLayers(root: Container<ChildNode>, model: Model) {
+	while (someAtRuleInTree(root, (node) => {
+		return node.nodes && someAtRuleInTree(node, (nested) => {
+			return nested.name === 'layer';
+		});
+	})) {
+		let foundUnexpectedLayerNesting = false;
+
+		root.walkAtRules('layer', (layerRule) => {
+			if (layerRule.parent === root) {
+				return;
+			}
+
+			if (layerRule.parent.type === 'atrule' && (layerRule.parent as AtRule).name === 'layer') {
+				const parent = layerRule.parent as AtRule;
+
+				model.layerNameParts.set(`${parent.params}.${layerRule.params}`, [...model.layerNameParts.get(parent.params), ...model.layerNameParts.get(layerRule.params)]);
```

Comment:
what is this line doing?

Reply:
Here we concatenate the parent and current layer params for these cases :

```css
@layer A {
  @layer C.D {}
}

@layer A.B {
  @layer C {}
}

@layer A.B {
  @layer C.D {}
}
```

We do not want to re-parse these but we still want the combined information.

Reply:
Here we concatenate the parent and current layer params for these cases :

```css
@layer A {
  @layer C.D {}
}

@layer A.B {
  @layer C {}
}

@layer A.B {
  @layer C.D {}
}
```

We do not want to re-parse these but we still want the combined information.